.github/workflows/cnb.yml

```diff
@@ -0,0 +1,99 @@
+name: CNB
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '00 4 * * 1-5'
+
+defaults:
+  run:
+    # Setting an explicit bash shell ensures GitHub Actions enables pipefail mode too,
+    # rather than only error on exit (improving failure UX when pipes are used). See:
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell
+    shell: bash
+
+permissions:
+  actions: write
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-cnb:
+    name: Update .NET CNB version
+    runs-on: pub-hk-ubuntu-24.04-ip
+
+    steps:
+      - uses: actions/create-github-app-token@v1
+        id: generate-token
+        with:
+          app-id: ${{ vars.LINGUIST_GH_APP_ID }}
+          private-key: ${{ secrets.LINGUIST_GH_PRIVATE_KEY }}
+
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          token: ${{ steps.generate-token.outputs.token }}
+
+      - name: Get next release and update files
```

Comment:
(optional)

I'm curious about why this updates only one version at a time? I'm presuming the main reason a version might have been skipped was that there was a classic-only regression (eg caught on the Hatchet tests in this repo), in which case you'd not want to release the interim version anyway (and in which case an "always update to latest CNB version" approach might be preferred?).

Anyway, I imagine you've thought a lot about the trade-offs, and you'll be the one using this automation - so any approach is fine with me :-)

Reply:
To be honest, I probably haven't given it as much thought as you may imagine :) It mostly came down to:

1. The classic buildpack should, ideally, be compatible with every CNB release. Bumping the CNB version one version at a time ensures that any changes required to support a new CNB release will be included in the PR that bumps the CNB version - with the goal of making each PR merge commit on the main branch focused and self-explanatory given the auto-merged changelog entries. 

    For example, when I merge https://github.com/heroku/buildpacks-dotnet/pull/157, I'll likely release a new version of the CNB. Running this workflow will create a PR that won't pass the Hatchet tests in this repo, as the CNB output will change in that version - so updating the CNB will require updating this repo to accommodate that, and just that.
2. That's the ideal scenario. I suppose a classic-only regression caused by a CNB release could perceivably not be worth the effort to address (if only to ensure a clean upgrade history like this). For those cases, we can of course just manually submit a PR bumping the pinned CNB version, skipping an incompatible CNB release.
3. It's possible this approach will prove to be unnecessarily impractical. I suspect it'll work just fine for the vast majority of updates (which will now be automated) though :)